src/lib.rs

```diff
@@ -309,6 +323,24 @@ impl From<&MarkedFile> for FileChange {
     }
 }
 
+pub fn validate_config(config: &GenerationConfig) -> Result<()> {
+    const VALID_BACKENDS: [&str; 3] = [
+        "diesel::pg::Pg",
+        "diesel::sqlite::Sqlite",
+        "diesel::mysql::Mysql",
+    ];
+
+    if !VALID_BACKENDS.contains(&config.diesel_backend.as_str()) {
+        return Err(Error::new(ErrorEnum::InvalidGenerationConfig(format!(
+            "Invalid diesel_backend '{}', please use one of the following: {:?}",
+            &config.diesel_backend,
+            VALID_BACKENDS.clone().join(", ")
+        ))));
+    }
+
+    Ok(())
+}
+
```

Comment:
why have backend validation against a specific list now when

> `- or, your custom diesel backend type (struct which implements `diesel::backend::Backend`)`

is still said and possible?

Reply:
thank you, good catch!